doc/dev/pylint_checking.md

```diff
@@ -0,0 +1,71 @@
+
+In the Azure SDK for Python repository, in addition to the standard pylint library, we added custom checkers that help to align our libraries with the [Azure SDK for Python Guidelines](https://guidelinescollab.github.io/azure-sdk/python_introduction.html).
+
+### How to run Pylint?
+
+One way to run pylint is to run at the package level with tox:
+
+    .../azure-sdk-for-python/sdk/eventgrid/azure-eventgrid>tox -e pylint -c ../../../eng/tox/tox.ini
+
+If you don't want to use tox you can also install and run pylint on its own:
+
+ - If taking this approach, in order to run with the pylintrc formatting and the custom pylint checkers you must also install the custom checkers and `SET` the pylintrc path. 
+
+- More info on how to run the custom pylint checkers [here](https://github.com/Azure/azure-sdk-tools/blob/main/tools/pylint-extensions/pylint-guidelines-checker/README.md)
+
+        pip install pylint
+        pip install pylint-guidelines-checker --index-url="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/"
+
+        .../azure-sdk-for-python>SET PYLINTRC="./pylintrc"
+        .../azure-sdk-for-python>pylint ./sdk/eventgrid/azure-eventgrid
+
+    Note that you may see different errors if running a different version of pylint than the one in CI.
+
+
+# Ignoring Pylint Checkers
+
+There are three ways to ignore pylint checkers:
+- If there is a pylint checker within your SDK that you do not think applies, or that you wish to ignore for that specific scenario (i.e protected-access) you can locally ignore it on the specific line of the warning with the comment `# pylint:disable=protected-access`. If a pylint warning appears multiple times within a file you can ignore all related warnings in that file by putting the disable comment at the top of the file.
+- Another way to disable a pylint checker is by putting in under the `disable` list in the `pylintrc` file. This should only happen if the checker is giving too many false positives, or is not applicable to our general library. Any additions to the pylintrc file should be discussed beforehand. 
+- (Not Recommended) The last way to disable a checker is by ignoring the entire package in the [environment_exclusions.py](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/ci_tools/environment_exclusions.py) or as a temporary fix ignoring parts of the package via the pylintrc `ignore-paths` option. 
+        
+
+# Pylint Warnings and Where to Find Them 
+
+Info on the custom pylint checkers [here](https://github.com/Azure/azure-sdk-tools/tree/main/tools/pylint-extensions/azure-pylint-guidelines-checker).
+
+In addition to being a part of the CI, the custom pylint checkers are also integrated into ApiView and will show as warnings there. 
```

Comment:
if possible, might be nice to link to an example of what a warning in apiview would look like or maybe add a screenshot here